File: poc-kronos-frontend/src/model/curso/Curso.js
class Curso {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.nome = dados ? dados.nome : null;
    this.sigla = dados ? dados.sigla : null;
    this.tituloMasculino = dados ? dados.tituloMasculino : null;
    this.tituloFeminino = dados ? dados.tituloFeminino : null;
    this.dataInicioVigencia = dados ? dados.dataInicioVigencia : null;
    this.dataFinalVigencia = dados ? dados.dataFinalVigencia : null;
    this.dataLimiteConclusao = dados ? dados.dataLimiteConclusao : null;
    this.tempoMaximoConclusao = dados ? dados.tempoMaximoConclusao : null;

    this.tempoMaximoDescontinuo = dados ? dados.tempoMaximoDescontinuo : null;
    this.cargaHoraria = dados ? dados.cargaHoraria : null;
    this.notaMinimaAprovacaoDireta = dados
      ? dados.notaMinimaAprovacaoDireta
      : null;
    this.notaMinimaAprovacaoRecuperacao = dados
      ? dados.notaMinimaAprovacaoRecuperacao
      : null;

    this.ativo = dados ? dados.ativo : null;
    this.idNivel = dados ? dados.idNivel : null;
    this.nomeNivel = dados ? dados.nomeNivel : null;
    this.idTipoRegimeLetivo = dados ? dados.idTipoRegimeLetivo : null;
    this.nomeTipoRegimeLetivo = dados ? dados.nomeTipoRegimeLetivo : null;

    this.idTipoTurno = dados ? dados.idTipoTurno : null;
    this.nomeTipoTurno = dados ? dados.NomeTipoTurno : null;

    this.idInstituicao = dados ? dados.idInstituicao : null;

    this.ativo = dados ? dados.ativo : null;
    this.idTipoMatrizHorario = dados ? dados.idTipoMatrizHorario : null;
    this.nomeTipoMatrizHorario = dados ? dados.nomeTipoMatrizHorario : null;
  }

  static validations(required, maxLength, numeric, dataFimMenor) {
    return {
      nome: { required, maxLength: maxLength(100) },
      sigla: { required, maxLength: maxLength(15) },
      tituloMasculino: { required, maxLength: maxLength(100) },
      tituloFeminino: { required, maxLength: maxLength(100) },
      dataInicioVigencia: { required },
      dataFinalVigencia: { dataFimMenor: dataFimMenor("dataInicioVigencia") },
      tempoMaximoConclusao: { required, numeric },
      tempoMaximoDescontinuo: { required, numeric },
      cargaHoraria: { required, numeric },
      notaMinimaAprovacaoDireta: { required, numeric },
      notaMinimaAprovacaoRecuperacao: { required, numeric },
      idNivel: { required },
      idTipoRegimeLetivo: { required },
      idTipoTurno: { required },
      idInstituicao: { required },
      idTipoMatrizHorario: { required }
    };
  }

  static validationMessages() {
    return {
      nome: {
        required: "Nome ?? obrigat??rio",
        maxLength: "Nome s?? aceita no m??ximo 100 caracteres"
      },
      sigla: {
        required: "Sigla ?? obrigat??ria",
        maxLength: "Sigla s?? aceita no m??ximo 15 caracteres"
      },
      tituloMasculino: {
        required: "T??tulo Masculino ?? obrigat??rio",
        maxLength: "T??tulo Masculino s?? aceita no m??ximo 100 caracteres"
      },
      tituloFeminino: {
        required: "T??tulo Feminino ?? obrigat??rio",
        maxLength: "T??tulo Feminino s?? aceita no m??ximo 100 caracteres"
      },
      dataInicioVigencia: {
        required: "Data Inicio Vig??ncia ?? obrigat??rio"
      },
      dataFinalVigencia: {
        dataFimMenor:
          "Data Final de Vig??ncia n??o pode ser menor que a Data In??cio de Vig??ncia"
      },
      tempoMaximoConclusao: {
        required: "Tempo M??ximo de Conclus??o ?? obrigat??rio",
        numeric: "Somente n??mero"
      },
      tempoMaximoDescontinuo: {
        required: "Tempo M??ximo de Descontinue ?? obrigat??rio",
        numeric: "Somente n??mero"
      },
      cargaHoraria: {
        required: "Carga Hor??ria ?? obrigat??ria",
        numeric: "Somente n??mero"
      },
      notaMinimaAprovacaoDireta: {
        required: "Nota m??nima de aprova????o direta ?? obrigat??ria",
        numeric: "Somente n??mero"
      },
      notaMinimaAprovacaoRecuperacao: {
        required: "Nota m??nima de aprova????o recupera????o ?? obrigat??ria",
        numeric: "Somente n??mero"
      },
      idNivel: {
        required: "N??vel ?? obrigat??rio"
      },
      idTipoRegimeLetivo: {
        required: "Tipo de Regime Letivo ?? obrigat??rio"
      },
      idTipoTurno: {
        required: "Tipo de Turno ?? obrigat??rio"
      },
      idInstituicao: {
        required: "Institui????o ?? obrigat??ria"
      },
      idTipoMatrizHorario: {
        required: "Tipo Matriz Hor??rio ?? obrigat??ria"
      }
    };
  }
}
export default Curso;

File: src/main/java/br/com/kronos/backend/aplicacao/curso/Curso.java
package br.com.kronos.backend.aplicacao.curso;


import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Curso { 
	
	public static final String COMBO_CACHE_NAME = "combo_cursos";
	public static final String COMBO_CACHE_NAME_PERFIL_PROFESSORES = "combo_cursos_perfil_professores";
    
    private long id;
    private String nome;
    private String sigla;
    private String tituloMasculino;
    private String tituloFeminino;
    private LocalDate dataInicioVigencia;
    private LocalDate dataFinalVigencia;
    private LocalDate dataLimiteConclusao;
    private int tempoMaximoConclusao;
    private int tempoMaximoDescontinuo;
    private int cargaHoraria;
    private Double notaMinimaAprovacaoDireta;
    private Double notaMinimaAprovacaoRecuperacao;
    private boolean ativo;
    private Integer idNivel;
    private int idTipoRegimeLetivo;
    private int idTipoTurno;
    private Integer idInstituicao;
    private Integer idTipoMatrizHorario;
    

	@Builder
	public Curso(long id, String nome, String sigla, String tituloMasculino, String tituloFeminino, LocalDate dataInicioVigencia, 
                 LocalDate dataFinalVigencia, LocalDate dataLimiteConclusao, int tempoMaximoConclusao, int tempoMaximoDescontinuo, 
                 int cargaHoraria, Double notaMinimaAprovacaoDireta, Double notaMinimaAprovacaoRecuperacao, boolean ativo, 
                 Integer idNivel, int idTipoRegimeLetivo, int idTipoTurno, Integer idInstituicao, Integer idTipoMatrizHorario) {

        this.id = id;
        this.nome = exigirTamanho(nome, "Nome", 1, 100);
        this.sigla = exigirTamanho(sigla, "Sigla", 1, 15);
        this.tituloMasculino = exigirTamanho(tituloMasculino, "Título masculino", 1, 100);
        this.tituloFeminino = exigirTamanho(tituloFeminino, "Título feminino", 1, 100);
        this.dataInicioVigencia = exigirNaoNulo(dataInicioVigencia, "Datade início de vigência");
        this.dataFinalVigencia = dataFinalVigencia;
        this.dataLimiteConclusao = dataLimiteConclusao;
        this.tempoMaximoConclusao = exigirNaoNulo(tempoMaximoConclusao, "Tempo máximo de conclusão"); 
        this.tempoMaximoDescontinuo = exigirNaoNulo(tempoMaximoDescontinuo, "Tempo máximo descontinuo");
        this.cargaHoraria = exigirNaoNulo(cargaHoraria, "Carga horária"); 
        this.notaMinimaAprovacaoDireta = exigirNaoNulo(notaMinimaAprovacaoDireta, "Nota mínima para aprovação direta");
        this.notaMinimaAprovacaoRecuperacao = exigirNaoNulo(notaMinimaAprovacaoRecuperacao, "Nota mínima para aprovação em recuperação");
        this.ativo = exigirNaoNulo(ativo, "Ativo");
        this.idNivel = exigirNaoNulo(idNivel, "Nível");
        this.idTipoRegimeLetivo = exigirNaoNulo(idTipoRegimeLetivo, "Tipo de regime letivo");
        this.idTipoTurno = exigirNaoNulo(idTipoTurno, "Tipo de turno");
        this.idInstituicao = exigirNaoNulo(idInstituicao, "Instituição");
        this.idTipoMatrizHorario = idTipoMatrizHorario;
       
	}
}
